helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -443,12 +469,19 @@ private void refreshBaseline(ResourceControllerDataProvider clusterData,
         getBaselineAssignment(_assignmentMetadataStore, currentStateOutput, resourceMap.keySet());
 
     // Read the best possible assignment from metadata store
-    Map<String, ResourceAssignment> currentBestPossibleAssignment = getBestPossibleAssignment(
-        _assignmentMetadataStore, currentStateOutput, resourceMap.keySet());
-
-    // Compute the new assignment
-    Map<String, ResourceAssignment> newAssignment = calculateAssignment(clusterData, clusterChanges,
-        resourceMap, activeNodes, currentBaseline, currentBestPossibleAssignment);
+    Map<String, ResourceAssignment> currentBestPossibleAssignment =
+        getBestPossibleAssignment(_assignmentMetadataStore, currentStateOutput,
```

Comment:
again, isn't it just duplicate since the below private method knows about the private field `_assignmentMetadataStore`, right?

Reply:
getBestPossibleAssignment contains more logic than just fetching the assignment from the store.